crates/grafbase-sdk/src/host_io/pubsub.rs

```diff
@@ -0,0 +1,18 @@
+//! Subscriber module providing interfaces for consuming field outputs from streams.
+//!
+//! This module defines the core `Subscriber` trait that abstracts over different
+//! implementations of subscribing to field output streams, allowing extensions to
+//! handle field outputs in a transport-agnostic way.
+
+use crate::{types::FieldOutput, Error};
+
+pub mod nats;
+
+/// A trait for subscribing to a stream of field outputs.
+pub trait Subscriber {
```

Comment:
Would have named it `Subscription`, subscriber for me is the client part.

Reply:
I agree